### ISSUE TITLE
onChange handler called on Enter and once again after debounceTimeout

### ISSUE THREAD

**Issue by isaacalves** [nkbt/react-debounce-input#64]
I'm using something like this:

`
	handleInputChange(e) {
		this.props.onInputChange(e.target.value);
	}
`

and on my render method:

`
       		<DebounceInput type='text'
       			onChange={ this.handleInputChange.bind(this) }
		        debounceTimeout={ 1000 }
       		/>
`

I see that `handleInputChange` is also called when I press Enter from the input field (although I could handle that myself in an `onSubmit` handler on the form). I assume that's intended, but why then does `handleInputChange` fires again after the timeout? 

Scenario:

1. focus on text field
2. type something and press Enter
3. handleInputChange() is called immediately
4. handleInputChange() is called again after the timeout

I see that I can use `forceNotifyByEnter` so it's not called on Enter key press. That solves the problem, but actually it would be nice to have it being called on Enter key press as well, as long as it clears the timeout.

**Comment by emhagman** [comment]
+1

**Comment by nkbt** [comment]
Makes sense. I would be happy to accept a PR on this

**Comment by hozefaj** [comment]
Let me give this is a shot. Since I am getting bugs filed on me for this issue.

**Comment by hozefaj** [comment]
Let me take a crack at fixing this.

**Comment by nkbt** [comment]
Thanks @hozefaj! Timeout should be cancelled on Enter, that feels like the most logical way.

**Comment by Data-Meister** [comment]
This is an issue with both notify on enter, and also notify on blur. 

The problem is [this line ](https://github.com/nkbt/react-debounce-input/blob/master/src/Component.js#L101)

`this.notify` is a method, but the code is attempting to access a property `this.notify.cancel` as if `this.notify` were an object

**Comment by nkbt** [comment]
That is ok, since lodash.debounce has .cancel method available ondebounced functions

**Comment by Data-Meister** [comment]
In that case, the problem is `this.notify` is not itself a debounced function. The debounced function is called from within `this.notify`

```
 const debouncedChangeFunc = debounce(event => {
        this.isDebouncing = false;
        this.doNotify(event);
      }, debounceTimeout);

      this.notify = event => {
        this.isDebouncing = true;
        debouncedChangeFunc(event);
      };
    }
```

**Comment by Data-Meister** [comment]
I've now fixed this bug, with code submitted as part of this PR

https://github.com/nkbt/react-debounce-input/pull/73

**Comment by nkbt** [comment]
Should be fixed in `react-debounce-input@3.0.1`
